test/junit/scala/concurrent/FutureTest.scala

```diff
@@ -112,4 +114,44 @@ class FutureTest {
     assertTrue(f.isCompleted)
     assertEquals(Some(Success(1)), f.value)
   }
+
+  @Test def t13058(): Unit = {
```

Comment:
the logic of `removeCallback` is non-trivial, maybe adding tests (either direct, or through a Promise) which verify:
* unregistering the first callback out of many
* last callback
* middle callback
would be a good idea? at each time verifying that the ordering of callbacks is preserved (if that's a property, that we want to guarantee, of course)